buildrunner/utils.py

```diff
@@ -309,3 +319,118 @@ def _cycle_colors(colors):
         colors[-1] = current
 
         return current
+
+def _acquire_flock(
+        lock_file: str,
+        logger: ContainerLogger,
+        mode: str,
+        timeout_seconds: float = LOCK_TIMEOUT_SECONDS,
+        exclusive: bool = True) -> io.IOBase:
+    """
+    Acquire file lock and open file with configurable timeout
+
+    :param lock_file: path and file name of file open and lock
+    :param logger: logger to log messages
+    :param mode: open mode
+    :param timeout_seconds: number of seconds for timeout
+    :param exclusive: config exclusive lock (True) or shared lock (False), defaults to True
+    :return: opened file object if successful else None
+    """
+    # Inspired by https://gist.github.com/jirihnidek/430d45c54311661b47fb45a3a7846537
+    flags = os.O_RDWR | os.O_CREAT
+
+    if 'a' in mode:
+        flags = flags | os.O_TRUNC
```

Comment:
Truncate the file with an append? That doesn't make much sense to me.

Reply:
Yeah I agree. I am not sure what I was thinking or why I have that there. I am going to relook at that function and the rest of this PR before this is merged.